doc: add minutes for meeting June 29 2022 (#1250)
nodejs/TSC: meetings/2022-06-29.md

* Richard Lau @richardlau (TSC)
* Robert Nagy @ronag (TSC)
* Michaël Zasso @targos (TSC)
* Rich Trott @Trott (TSC)

## Agenda

### Announcements

* No announcements this wek

### CPC and Board Meeting Updates

*Extracted from **tsc-agenda** labeled issues and pull requests from the **nodejs org** prior to the meeting.

* No updates this week.

### nodejs/node

* Move stream development out from core [#43520](https://github.com/nodejs/node/issues/43520)
  * Robert open the issue, there are advantages
    * Do bug fixes and get out to users faster
    * Seems to be working for undici
  * Matteo until recently external package was not up to date enough
  * James, one concern is rate of breaking changes. Do we need to have any
    processes to make sure any breaking changes are visible to TSC ?
  * Michael what are the main motivations
  * Robert
    * can get fixes out faster
    * avoid having modules getting out of date
  * Richard, how would we deal with the case that there are SemVer
    major differences between Node.js versions ?
  * Ruben, why different for streams?
  * Richard, if we encourage to use module versus what is in core, we might get less coverage

>     for what we have in core.  

Author: ```suggestion
    for what we have in core.
```

  * Robert not as much going on right now in stream, so can’t comment on value
    one way or other
  * Matteo very conflicted on top
    * Would simplify development, can get latest fixes out faster
    * On other hand, don’t think we have the workforce to meet the core guarantees etc.
    * not convinced it’s the right use of the development time that we have
    * Maybe see how we handle it once we bring undici into core and figure out how
      we manage and then consider how it would apply to other cases.

* Consider explicit string formatting module [#43382](https://github.com/nodejs/node/issues/43382)
  * Ruben added to agenda, had suggested new module to hold apis just related to formatting.
     Util is getting overloaded with lots of stuff.
    * had talked about needing more colorizing stuff in core, in part for use within core itself
    * Implemented a PR re-using the chalk API
    * Had reached out to one of the maintainers of Chalk
    * Concerns raised by one of the maintainers in terms of harm to existing module, lots of
      discussion in the issue itself.
    * Key question is how to work more closely together

>   * Matteo - Significant part of the friction with chalk today is around ESM. (Current Chalk versions only support ESM), we should be providing some guidance on this front.  Most downloaded version is still CJS version.  

Author: ```suggestion
  * Matteo - Significant part of the friction with chalk today is around ESM. (Current Chalk versions only support ESM), we should be providing some guidance on this front.  Most downloaded version is still CJS version.
```

  * Ruben actually 3
    * Do we want a module for all string formatting functionalites for existing APIs?
    * Should we vendor something like chalk or should we implement a
      similar functionality on our own?
    * Implement a process how we approach userland module authors in
      case we implement a functionality that exists in userland and is widely used

* tools: add `.git-blame-ignore-revs` file [#43017](https://github.com/nodejs/node/pull/43017)
  * Darshan had added to agend, don’t feel strongly at this point, so closed.
  * Nothing further to discuss at this point.

* build: add devcontainer configuration [#40825](https://github.com/nodejs/node/pull/40825)
  * Nothing further to discuss this week.

* Proposal: Node.js vendor namespace [#43413](https://github.com/nodejs/node/issues/43413)
  * Main thing is it needs more people to engage/discuss
  * Lots of tricky issues to, more specifics on proposal will help drive that discussion.

### nodejs/i18n

* I would like to become a member [#612](https://github.com/nodejs/i18n/issues/612)
  * Need more time than we have, defer longer discussion to next time
  * Today translations are done by people doing PRs to website

>   * Biggest limitation is that we can’t translate the APIs

Reviewer: ```suggestion
  * Big limitation is that we can’t translate the API docs
```

>   * Efforts to use Crowd in petered out

Reviewer: ```suggestion
  * Efforts to use Crowdin petered out; people who were running it are no longer active
```

  * Issue is people without current connection wanting to help
    * Issue is we have seen work peeter out
  * Do we want to try to encourage, or say “tried this” did not work. Longer conversation.

## Strategic Initiatives

* Ran out of time, skip for next time in general

* V8 Currency: currently unable to update to >=10.3 due to multiple issues:

>   * Windows (debug build): deps: update V8 to 10.3 node#43195 (comment)

Reviewer 2: ```suggestion
  * Windows (debug build): deps: update V8 to 10.3 [node#43195 (comment)](https://github.com/nodejs/node/pull/43195#issuecomment-1135452762)
```

>   * Linux (debug build): DCHECK crash on !i_isolate->is_execution_terminating() node-v8#227

Reviewer 2: ```suggestion
  * Linux (debug build): DCHECK crash on !i_isolate->is_execution_terminating() [node-v8#227](https://github.com/nodejs/node-v8/issues/227)
```

>   * macOS: Build error on osx node-v8#229 (there's a suggested fix in the issue, but be prepared for other problems with tests after the compilation is fixed)

Reviewer 2: ```suggestion
  * macOS: Build error on osx [node-v8#229](https://github.com/nodejs/node-v8/issues/229) (there's a suggested fix in the issue, but be prepared for other problems with tests after the compilation is fixed)
```

>   * test-hash-seed started to fail recently: test-hash-seed fails node-v8#231

Reviewer 2: ```suggestion
  * test-hash-seed started to fail recently: test-hash-seed fails [node-v8#231](https://github.com/nodejs/node-v8/issues/231)
```
